src/collection.jl

```diff
@@ -115,30 +194,198 @@ end
+
+Iterates over the `CCDData`s of the collection applying function `f` at each step.
+
+It returns an array of output values of function `f` applied on `CCDData`s.
+In addition to applying function `f`, the outputs can be saved. If `save = true`, it enables programmatical saving of returned value of the function `f` using [`CCDReduction.writefits`](@ref). File is saved at `path` specified by the user.
+Suffix and prefix can be added to filename of newly created files by modifying `save_suffix` and `save_prefix`, `save_delim` is used as delimiter.
+`ext` is the extension of files in collection, by default it is set to `r"fits(\\.tar\\.gz)?"i`.
+
+# Example
+```julia
+collection = fitscollection("~/data/tekdata")
+processed_images = map(ccds(collection)) do img
+    trim(img, (:, 1040:1059))
+end
+```
+The above generates `processed_images` which consists of trimmed versions of images present in `collection`.
+
+For saving the `processed_images` simultaneously with the operations performed
+```julia
+processed_images = map(ccds(collection; save = true, path = "~/data/tekdata", save_prefix = "trimmed")) do img
+    trim(img, (:, 1040:1059))
+end
+```
+The trimmed images are saved as `trimmed_(original_name)` (FITS files) at `path = "~/data/tekdata"` as specified by the user.
+
+Mapping version of `ccds` function is interfaced on iterative version of `images`, any valid parameter can be passed into iterative version as `kwargs`.
+"""
+function ccds(f, collection; save = any(!isnothing, (save_prefix, path, save_suffix)), path = nothing, save_prefix = nothing, save_suffix = nothing, save_delim = "_", ext = r"fits(\.tar\.gz)?"i, kwargs...)
+    image_iterator = ccds(collection; kwargs...)
+    locations = collection.path
+
+    processed_images = map(zip(locations, image_iterator)) do (location, output)
+        processed_image = f(output)
+        if save
+            save_path = generate_filename(location, path, save_prefix, save_suffix, save_delim, ext)
+            writefits(save_path, processed_image)
+        end
+        processed_image
+    end
+
+    return processed_images
+end
+
+
+"""
+    filenames(f, collection; save = any(!isnothing, (save_prefix, path, save_suffix)), path = nothing, save_prefix = nothing, save_suffix = nothing, save_delim = "_", ext = r"fits(\\.tar\\.gz)?"i, kwargs...)
+
+Iterates over the file paths of the collection applying function `f` at each step.
+
+It returns an array of output values of function `f` applied on file paths.
+In addition to applying function `f`, the outputs can be saved. If `save = true`, it enables programmatical saving of returned value of the function `f` using [`CCDReduction.writefits`](@ref). File is saved at `path` specified by the user.
+Suffix and prefix can be added to filename of newly created files by modifying `save_suffix` and `save_prefix`, `save_delim` is used as delimiter.
+`ext` is the extension of files in collection, by default it is set to `r"fits(\\.tar\\.gz)?"i`.
```

Comment:
```suggestion
The output from `f` can be saved using the appropriate keyword arguments. The `save_prefix` argument will add a prefix to each filename delimited by `save_delim`. `save_suffix` will add a suffix prior to the extension, which can be manually provided via `ext`, similar to [`fitscollection`](@ref). Files will be saved in the directory they are stored unless `path` is given. Finally, `save` will default to `true` if any of the previous arguments are set, but can be manually overridden (useful for testing). Files will be saved using [`CCDReduction.writefits`](@ref).
```

Reply:
> Files will be saved in the directory they are stored unless `path` is given

Should we implement this? Or should we leave this for user? (Haven't implemented this yet, as in rn one has to explicity mention the path)

PS: I don't think it would be a trouble for the user to go for `path = @__DIR__`
Thoughts?

Reply:
`@__DIR__` would be the same directory as where this file is, that is definitely not what we want. I think the two options we have are either the original path or the current working directory. I think the one that makes more sense  is to have path default to the same directory as the file